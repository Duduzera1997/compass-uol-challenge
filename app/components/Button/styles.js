import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  margin-top: 10%;
  align-items: center;
`;

export const TouchableButton = styled.TouchableOpacity`
  background-color: ${({ theme }) => theme.colors.primary.red};
  border-radius: 8px;
  align-items: center;
  justify-content: center;
  width: 70%;
  height: 50px;
`;

export const Label = styled.Text`
  font-family: ${({ theme }) => theme.fonts.primary.family}Bold;
  font-size: 18px;
  color: ${({ theme }) => theme.colors.secondary.white};
`;
