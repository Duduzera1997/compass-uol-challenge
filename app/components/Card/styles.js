import styled from 'styled-components/native';

export const ContainerCard = styled.View`
  height: 120px;
  align-items: center;
  padding-bottom: 10px;
`;

export const CardTouchable = styled.TouchableOpacity`
  background-color: ${({ theme }) => theme.colors.secondary.white};
  height: 100%;
  border-radius: 9px;
  flex-direction: row;
`;

export const CardContent = styled.View`
  flex: ${({ flex }) => flex};
  justify-content: flex-start;
  align-items: center;
  padding: 5px 5px 2px 7px;
`;

export const PreviewImage = styled.Image`
  width: 100%;
  height: 95%;
  border-radius: 10px;
`;

export const Text = styled.Text.attrs({
  numberOfLines: 2,
})`
  width: 99%;
  font-size: ${({ size }) => size}px;
  font-family: ${({ theme, fontBold }) =>
    `${theme.fonts.primary.family}${fontBold ? 'Bold' : 'Regular'}`};
  color: ${({ theme }) => theme.colors.secondary.darkGrey};
  padding: 4px 2px 2px 4px;
`;

export const TextDate = styled(Text)`
  padding: 0 2px 2px 4px;
  color: ${({ theme }) => theme.colors.secondary.grey};
`;
