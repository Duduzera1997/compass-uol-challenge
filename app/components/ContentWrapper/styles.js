import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  padding-top: 10px;
`;

export const ContainerByFlex = styled.View`
  flex: ${({ flex }) => flex};
`;
