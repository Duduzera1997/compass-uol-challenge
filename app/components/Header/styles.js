import styled from 'styled-components/native';

export const ContainerHeader = styled.View`
  flex: 1;
  align-items: flex-start;
`;

export const HeaderTitle = styled.Text`
  font-size: 27px;
  font-family: ${({ theme }) => theme.fonts.primary.family}Bold;
  color: ${({ theme }) => theme.colors.secondary.white};
  padding-top: 15px;
  padding-left: 15px;
  letter-spacing: 2px;
`;

export const DateLabel = styled.Text`
  font-size: 14px;
  font-family: ${({ theme }) => theme.fonts.primary.family}Regular;
  color: ${({ theme }) => theme.colors.secondary.white};
  padding-top: 2px;
  padding-left: 16px;
`;
