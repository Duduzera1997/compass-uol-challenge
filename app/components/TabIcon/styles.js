import styled from 'styled-components/native';
import Feather from 'react-native-vector-icons/Feather';

export const ContainerIcon = styled.View`
  justify-content: center;
  align-items: center;
`;
export const IconTab = styled(Feather)`
  font-size: ${({ size }) => size}px;
`;
