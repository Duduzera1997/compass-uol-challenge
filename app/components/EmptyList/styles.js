import styled from 'styled-components/native';
import { white } from '~/styles/colors';

export const ContainerEmptyList = styled.View`
  flex: 1;
  margin-top: 50%;
  justify-content: center;
  align-items: center;
`;

export const Label = styled.Text`
  font-size: 22px;
  font-family: ${({ theme }) => theme.fonts.primary.family}Regular;
  color: ${({ theme }) => theme.colors.secondary.white};
  text-align: center;
`;

export const ActivityIndicator = styled.ActivityIndicator.attrs({
  size: 'small',
  color: white,
})``;
