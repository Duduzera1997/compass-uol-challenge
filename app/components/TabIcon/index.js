import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';
import { white, whiteLight } from '~/styles/colors';

function TabIcon({ icon, focused }) {
  return (
    <SC.ContainerIcon testID="component-tabIcon-container">
      <SC.IconTab
        name={icon}
        size={20}
        color={focused ? white : whiteLight}
        testID="component-tabIcon-icon"
      />
    </SC.ContainerIcon>
  );
}

TabIcon.defaultProps = {
  icon: 'airplay',
  focused: false,
};

TabIcon.propTypes = {
  icon: PropTypes.string,
  focused: PropTypes.bool,
};

export default TabIcon;
