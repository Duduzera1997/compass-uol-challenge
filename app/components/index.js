import TabIcon from './TabIcon';
import LinearBackground from './LinearBackground';
import Loading from './Loading';
import Header from './Header';
import Card from './Card';
import ListHelper from './ListHelper';
import ListArticles from './ListArticles';
import ViewArticleModal from './ViewArticleModal';
import Button from './Button';
import SearchInput from './SearchInput';
import ContentWrapper from './ContentWrapper';

export {
  TabIcon,
  LinearBackground,
  Loading,
  Header,
  Card,
  ListHelper,
  ListArticles,
  ViewArticleModal,
  Button,
  SearchInput,
  ContentWrapper,
};
