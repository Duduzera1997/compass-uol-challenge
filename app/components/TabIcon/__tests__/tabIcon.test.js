import React from 'react';
import { render } from '@testing-library/react-native';

import { TabIcon } from '~/components';
import { red, grey } from '~/styles/colors';

describe('TabIcon Component', () => {
  it('should be render container in center', () => {
    const { getByTestId, toJSON } = render(<TabIcon />);
    const container = getByTestId('component-tabIcon-container');
    expect(container).toHaveStyleRule(
      'justify-content',
      'center',
      'align-items',
      'center',
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should be render a TabIcon without props', () => {
    const { getByTestId, toJSON } = render(<TabIcon />);
    const icon = getByTestId('component-tabIcon-icon');
    expect(icon).toHaveStyleRule('font-size', 20);
    expect(icon).toHaveStyleRule('color', grey);
    expect(toJSON()).toMatchSnapshot();
  });
  it('should be render a TabIcon with focused prop', () => {
    const { getByTestId, toJSON } = render(<TabIcon focused />);
    const icon = getByTestId('component-tabIcon-icon');
    expect(icon).toHaveStyleRule('font-size', 20);
    expect(icon).toHaveStyleRule('color', red);
    expect(toJSON()).toMatchSnapshot();
  });
});
