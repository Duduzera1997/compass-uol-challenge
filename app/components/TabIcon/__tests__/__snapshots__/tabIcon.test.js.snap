// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabIcon Component should be render a TabIcon with focused prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  testID="component-tabIcon-container"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
        },
        Array [
          Object {
            "fontSize": 20,
          },
        ],
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="component-tabIcon-icon"
  >
    
  </Text>
</View>
`;

exports[`TabIcon Component should be render a TabIcon without props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  testID="component-tabIcon-container"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "rgba(255,255,255,0.5)",
          "fontSize": 20,
        },
        Array [
          Object {
            "fontSize": 20,
          },
        ],
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="component-tabIcon-icon"
  >
    
  </Text>
</View>
`;

exports[`TabIcon Component should be render container in center 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  testID="component-tabIcon-container"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "rgba(255,255,255,0.5)",
          "fontSize": 20,
        },
        Array [
          Object {
            "fontSize": 20,
          },
        ],
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="component-tabIcon-icon"
  >
    
  </Text>
</View>
`;
